dagsync/subscriber.go

```diff
@@ -169,6 +170,7 @@ func NewSubscriber(host host.Host, ds datastore.Batching, lsys ipld.LinkSystem,
 		}
 		dtSync, err = dtsync.NewSyncWithDT(host, opts.dtManager, opts.graphExchange, &lsys, blockHook)
 	} else {
+		ds := namespace.Wrap(ds, datastore.NewKey("data-transfer-v2"))
```

Comment:
Doing this will have the effect of adding the prefix `"data-transfer-v2"` to all new keys stored in the datastore, and will make all the existing keys inaccessible.  All of the other items stored in the datastore, that we care about keeping, already have their own prefixes.  Perhaps it would be better to delete all the keys that do not have one of said prefixes, to avoid migration. ... or am I missing something about what this change actually does?

Reply:
>to all new keys stored in the datastore 

_For data transfer manager only_. Note that DAGs are stored in the linksystem that's passed in. 

As for other keys, that are storetheindex specific, they should continye to be accessibl. Unless I have missed something.

Reply:
If we are OK with leaving FSM state garbage behind, then this will get our nodes working again.

Reply:
https://github.com/ipni/storetheindex/issues/1380